src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -131,9 +132,9 @@ runLanguageServer options userHandlers onInitialConfig onConfigChange getIdeStat
     where
         handleInit :: IO () -> (LspId -> IO ()) -> (LspId -> IO ()) -> Chan (Message config) -> LSP.LspFuncs config -> IO (Maybe err)
         handleInit exitClientMsg clearReqId waitForCancel clientMsgChan lspFuncs@LSP.LspFuncs{..} = do
-
+            cfg <- config
             ide <- getIdeState getNextReqId sendFunc (makeLSPVFSHandle lspFuncs) clientCapabilities
-                               withProgress withIndefiniteProgress
+                               withProgress withIndefiniteProgress cfg
```

Comment:
I think the rootUri is available through `LspFuncs` as `rootPath`, which is a bit confusingly admittedly. Can we just pass that here instead? Instead of passing through the config.